app/src/renderer/js/preferences.js

```diff
@@ -94,6 +94,50 @@ document.addEventListener('DOMContentLoaded', () => {
     $j('#plugins-installed').html(html);
   }
 
+  function loadDefaultShareService() {
+    const shareServices = plugins.getShareServices();
+    const template = `
+      <select>
+        <option>Show editor [default]</option>
+        <% _.forEach(options, option => { %>
+            <option <%- option.selected ? 'selected' : '' %> value='<%- option.value %>'>
+              <%- option.title %> [<%- option.format %>]
+            </option>
+        <% }); %>
+      </select>
+    `;
+
+    const compiled = _.template(template);
+    const current = app.kap.settings.get('defaultShare');
+
+    const options = shareServices.reduce((all, service) => {
```

Comment:
Could you rewrite this using `for..of`? `reduce()` usage isn't that readable and easy to follow, so I'd just populate a results array.